### DISCUSSION TITLE
[Feature] TypeScript support

### DISCUSSION THREAD

**Discussion by bakcxoj** [bridge-core/bridge.#162] [Ideas]
**Summary**
TypeScript is a great language for interacting with JavaScript APIs. I see that bridge. is made with TypeScript so I hope you'd agree.

This could be achieved by using the TypeScript npm module. When a file is saved it can be compiled to JavaScript and saved in a separate file.

**Comment by solvedDev** [comment]
Do you want TypeScript support for Minecraft's scripting API?

**Comment by bakcxoj** [comment]
Well, that would be nice, but I'm talking about a way to both use bridge and TS, as right now it seems you have to choose.

See the generator-minecraft-addon generator for yeoman

**Comment by solvedDev** [comment]
I can confirm that we're actively looking into supporting TS for all Bridge APIs soon. 
Support for Minecraft's scripting API will have to wait until we support actually compiling projects
